### ISSUE TITLE
The `/etc/letsencrypt/live/npm-$/` path changes from time to time, resulting in unusable SSL feature of the addon

### ISSUE THREAD

**Issue by reenberg** [hassio-addons/addon-nginx-proxy-manager#145]
# Problem/Motivation
The addon is spewing exceptions in the logs, and all SSL functionality in the app is not working. 

Navigating to the "SSL Certificates" tab yield sa big yellow box with the text "404" and a "try again" button. 

Any configured hosts with SSL is showing a yellow "warning" status.

## Expected behavior

Fully functioning app, and no errors in the logs.

## Actual behavior

The logs is spewing the following error:

```
nginx: [emerg] cannot load certificate "/etc/letsencrypt/live/npm-1/fullchain.pem": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/letsencrypt/live/npm-1/fullchain.pem','r') error:2006D080:BIO routines:BIO_new_file:no such file)
```

When listing the content of the `live` directory, I can see that I have a `npm-2` folder instead.  It seems that this npm folder changes under some special situations?  

I tried to restart the container, but that didn't change the path that it tried to use.  I can see from other bug reports, that they are refering to `npm-7`, `npm-9`, etc folders, so it do change somehow.

Moving the folder from `npm-2` to the currently expected `npm-1` name resolved the issues.

I believe that this started to be an issue when I upgraded the Supervisor to the latest version.  However I'm not really sure how that could affect anything, especially when restarting the container doesn't seem to change the path that it tries to use.

## Steps to reproduce

As I don't know the reason for the expected path to change, I can't 100% say how to reproduce, besides try renaming the folder to another number, and then the failures of the app will be reproducible.

There are quite a few old reports that seems to also be referring to this issue, however they have all been closed without an actual fix.

**Comment by reenberg** [comment]
mmm apparently I was a bit too quick with that perceived fix...

Now certbot fails to renew the certificate, as the letsencrypt config still tried

```
[1/30/2021] [4:21:18 PM] [SSL      ] › ✖  error     Error: Command failed: /usr/bin/certbot renew --non-interactive --quiet --config "/etc/letsencrypt.ini" --preferred-challenges "dns,http" --disable-hook-validation  
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/certbot/_internal/renewal.py", line 63, in _reconstitute
    renewal_candidate = storage.RenewableCert(full_path, config)
  File "/usr/lib/python3.8/site-packages/certbot/_internal/storage.py", line 466, in __init__
    self._check_symlinks()
  File "/usr/lib/python3.8/site-packages/certbot/_internal/storage.py", line 532, in _check_symlinks
    raise errors.CertStorageError(
certbot.errors.CertStorageError: expected /etc/letsencrypt/live/npm-2/cert.pem to be a symlink   <--  certbox still uses npm-2 folder.
Renewal configuration file /etc/letsencrypt/renewal/npm-2.conf is broken. Skipping.
0 renew failure(s), 1 parse failure(s)
    at ChildProcess.exithandler (child_process.js:303:12)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
`QueryBuilder#allowEager` method is deprecated. You should use `allowGraph` instead. `allowEager` method will be removed in 3.0
`QueryBuilder#eager` method is deprecated. You should use the `withGraphFetched` method instead. `eager` method will be removed in 3.0
QueryBuilder#omit is deprecated. This method will be removed in version 3.0
Duplicate relation "access_list" in a relation expression. You should use "a.[b, c]" instead of "[a.b, a.c]". This will cause an error in objection 2.0
``` 

Figuring out that `npm-2` apparently is the "name" of my certificate, created by the addon, starts to make sense.  I initially created test certificate in the SSL tab, which I replaced.  
It may seem that some reference the the first certificate was still present, even after I deleted it under the SSL tab.  

I moved the folder back, after the addon had started, which made certbot happy.  I then restarted the addon a few times, and there doesn't seem to be any references to the `npm-1` certificate anymore.   

Anyways, it still seems like a bug, that it doesn't 100% removes all references when deleting a certificate, and thus it may utterly break when restarting the app.

**Comment by loganwedwards** [comment]
Adding another data point that I am experiencing the exact same behavior. I have 2 directories under `live`, `npm-2` and `npm-3` (I assume since I have 2 certificates). This has rendered my SSL unusable. If I can provide any further debug to help, let me know.

**Comment by loganwedwards** [comment]
Ok, so I found some manual workarounds and the root issues:

By using the official `phpMyAdmin` addon, the `proxy_host` table shows the records for each added host. The host in question still had a reference to the old/deleted certificate, so I updated the `id` field to what I expected. I saw that the addon has a `reset_database` option, but was not sure if it was really appropriate for this case, so I did not use it.

Further, ssh into the docker container `docker exec -it {id} /bin/bash`, I found the offending configuration file named `/data/nginx/proxy_host/1.conf`. Edit the `ssl_certificate` and `ssl_certificate_key` lines to point to the correct `npm-#` certificate. In my case, `npm-3`. Further, I renamed the file from `1.conf` to `3.conf` and restarted the addon. No further errors.

As @reenberg mentions, it sounds like the addon might not be properly cleaning orphaned data. In my case, I created the hosts a while back, but made some changes to the attached certificates. I am guessing that when I deleted the certificate attached to the host is what caused issues. 

I hope this helps others!
